libs/web-components/src/components/badge/Badge.svelte

```diff
@@ -308,4 +357,141 @@
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-badge-default-subtle-color-border);
+    color: var(--goa-badge-default-subtle-color-content);
+  }
+
+  .v2.goa-badge.badge-archived.badge-subtle {
+    background-color: var(--goa-badge-archived-subtle-color-bg);
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-badge-archived-subtle-color-border);
+    color: var(--goa-badge-archived-subtle-color-content);
+  }
+
+  /* Version 2: Extended Colours */
+
+  .v2.badge-sky {
+    background-color: var(--goa-color-extended-sky-default);
+    color: var(--goa-color-extended-sky-text);
+  }
+
+  .v2.badge-prairie {
+    background-color: var(--goa-color-extended-prairie-default);
+    color: var(--goa-color-extended-prairie-text);
+  }
+
+  .v2.badge-lilac {
+    background-color: var(--goa-color-extended-lilac-default);
+    color: var(--goa-color-extended-lilac-text);
+  }
+
+  .v2.badge-pasture {
+    background-color: var(--goa-color-extended-pasture-default);
+    color: var(--goa-color-extended-pasture-text);
+  }
+
+  .v2.badge-sunset {
+    background-color: var(--goa-color-extended-sunset-default);
+    color: var(--goa-color-extended-sunset-text);
+  }
+
+  .v2.badge-dawn {
+    background-color: var(--goa-color-extended-dawn-default);
+    color: var(--goa-color-extended-dawn-text);
+  }
+
+  .v2.badge-subtle.badge-sky {
+    background-color: var(--goa-color-extended-sky-light);
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-color-extended-sky-default);
+    color: var(--goa-color-extended-sky-text);
+  }
+
+  .v2.badge-subtle.badge-prairie {
+    background-color: var(--goa-color-extended-prairie-light);
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-color-extended-prairie-default);
+    color: var(--goa-color-extended-prairie-text);
+  }
+
+  .v2.badge-subtle.badge-lilac {
+    background-color: var(--goa-color-extended-lilac-light);
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-color-extended-lilac-default);
+    color: var(--goa-color-extended-lilac-text);
+  }
+
+  .v2.badge-subtle.badge-pasture {
+    background-color: var(--goa-color-extended-pasture-light);
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-color-extended-pasture-default);
+    color: var(--goa-color-text-default-pasture-text);
+  }
+
+  .v2.badge-subtle.badge-sunset {
+    background-color: var(--goa-color-extended-sunset-light);
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-color-extended-sunset-default);
+    color: var(--goa-color-text-default-sunset-text);
+  }
+
+  .v2.badge-subtle.badge-dawn {
+    background-color: var(--goa-color-extended-dawn-light);
+    box-shadow: inset 0 0 0 var(--goa-badge-border-width)
+      var(--goa-color-extended-dawn-default);
+    color: var(--goa-color-text-default-dawn-text);
```

Comment:
This should reference the extended colour token, not text-default.
```suggestion
    color: var(--goa-color-extended-dawn-text);
```